engine/src/main/java/io/camunda/zeebe/streamprocessor/ProcessingStateMachine.java

```diff
@@ -179,6 +179,7 @@ public ProcessingStateMachine(
 
   private void skipRecord() {
     notifySkippedListener(currentRecord);
+    inProcessing = false;
```

Comment:
:question: Good catch. Out of curiosity. Did you encounter an issue or how did you find it?

Reply:
The newly added test failed because the processors return EMPTY processing result.